src/glue/jobs/run_great_expectations_on_parquet.py

```diff
@@ -0,0 +1,408 @@
+    context: "EphemeralDataContext",
+    expectation_suite_name: str,
+    validation_result: Dict[str, str],
+    batch_identifier: RuntimeBatchRequest,
+    run_identifier: RunIdentifier,
+) -> "EphemeralDataContext":
+    """Adds the validation results manually to the validation store.
+        This is a workaround for a EphemeralDataContext context object,
+        and for us to avoid complicating our folder structure to include
+        checkpoints/other more persistent data context object types
+        until we need that feature
+
+    Args:
+        context (EphemeralDataContext): context object to add results to
+        expectation_suite_name (str): name of expectation suite
+        validation_result (Dict[str, str]): results outputted by gx
+            validator to be stored
+        batch_identifier (RuntimeBatchRequest): metadata containing details of
+            the batch request
+        run_identifier (RunIdentifier): metadata containing details of the gx run
+
+    Returns:
+        EphemeralDataContext: context object with validation results added to
+    """
+    expectation_suite = context.get_expectation_suite(expectation_suite_name)
+    # Create an ExpectationSuiteIdentifier
+    expectation_suite_identifier = ExpectationSuiteIdentifier(
+        expectation_suite_name=expectation_suite.expectation_suite_name
+    )
+
+    # Create a ValidationResultIdentifier using the run_id, expectation suite, and batch identifier
+    validation_result_identifier = ValidationResultIdentifier(
+        expectation_suite_identifier=expectation_suite_identifier,
+        batch_identifier=batch_identifier,
+        run_id=run_identifier,
+    )
+
+    context.validations_store.set(validation_result_identifier, validation_result)
+    return context
+
+
+def main():
+    args = read_args()
+    run_id = RunIdentifier(run_name=f"run_{datetime.now().strftime('%Y%m%d_%H%M%S')}")
+    expectation_suite_name = f"{args['data_type']}_expectations"
+    s3 = boto3.client("s3")
+    context = create_context(
+        s3_bucket=args["shareable_artifacts_bucket"],
+        namespace=args["namespace"],
+        key_prefix=f"great_expectation_reports/{args['data_type']}/parquet/",
+    )
+    glue_context = GlueContext(SparkContext.getOrCreate())
+    logger.info("get_spark_df")
+    spark_df = get_spark_df(
+        glue_context=glue_context,
+        parquet_bucket=args["parquet_bucket"],
+        namespace=args["namespace"],
+        data_type=args["data_type"],
+    )
+    logger.info("get_batch_request")
+    batch_request = get_batch_request(spark_df, args["data_type"], run_id)
+    logger.info("add_expectations")
+
+    # Load the JSON file with the expectations
+    logger.info("reads_expectations_from_json")
+    expectations_data = read_json(
+        s3=s3,
+        s3_bucket=args["cfn_bucket"],
+        key_prefix=args["expectation_suite_key_prefix"],
+    )
+    logger.info("adds_expectations_from_json")
+    add_expectations_from_json(
```

Comment:
In an ideal world, rather than logging the error instead of raising it, only deploy the GX job and associated trigger for a data type if that data type has expectations. That would require us to do some Jinja on `src/glue/resources/data_values_expectations.json` (sort of like how we do Jinja on `src/glue/resources/table_columns.json` in `templates/glue-tables.j2`). But if that's too complicated then we can log the error and return early.

Reply:
Yup it looks very doable. I'll add this and [that](https://github.com/Sage-Bionetworks/recover/pull/139#discussion_r1747794348) in.